compat/mingw.c

```diff
@@ -661,8 +661,18 @@ static int mingw_open_append(wchar_t const *wfilename, int oflags, ...)
 	handle = CreateFileW(wfilename, FILE_APPEND_DATA,
```

Comment:
The commit message needs a little bit of work... Just imagine that you yourself stumble over this commit in six months from now, not remembering what you did, but wanting to figure it out.

To that end, the Git project tries to enforce very informative commit messages, [in a somewhat idiomatic form](https://git-scm.com/docs/SubmittingPatches#describe-changes).

In particular, the commit message should focus a bit more on the "Why?" and to a lesser extent the "What?" than the "How?".

In this case, I would strongly suggest something along those lines:

```
mingw: cope with the Isilon network file system

On certain network filesystems (*cough* Isilon), when the directory in
question is missing, `raceproof_create_file()` fails with an
`ERROR_INVALID_PARAMETER` instead of an `ERROR_PATH_NOT_FOUND`.

Since it is highly unlikely that we produce such an error by mistake
(the parameters we pass are fairly benign), we can be relatively certain
that the directory is missing in this instance. So let's just translate
that error automagically.

This fixes https://github.com/git-for-windows/git/issues/1345.

Signed-off-by: ...
```

Reply:
I will reuse most of your example, if you are fine by that. I do not what to re-invent the wheel as we say in Dutch :). I personally only work on small scale projects, but I fully understand the more verbose commit.